yaserde_derive/src/de/expand_struct.rs

```diff
@@ -443,6 +444,11 @@ pub fn parse(
               #write_unused
               depth -= 1;
             }
+            XmlEvent::EndDocument => {
+              if #flatten {
+                break;
+              }
+            }
```

Comment:
This introduces an endless loop where previously an `unknown event` error was returned.